crates/dojo-core/src/interfaces.cairo

```diff
@@ -40,7 +42,13 @@ trait ISystem {
 trait IWorldFactory {
     fn set_world(class_hash: starknet::ClassHash);
     fn set_executor(class_hash: starknet::ClassHash);
-    fn spawn(name: ShortString, components: Array::<starknet::ClassHash>, systems: Array::<starknet::ClassHash>);
+    fn spawn(
+        name: ShortString,
+        components: Array::<starknet::ClassHash>,
+        systems: Array::<starknet::ClassHash>,
+        auth_components: Array::<starknet::ClassHash>,
+        auth_systems: Array::<starknet::ClassHash>
```

Comment:
Added auth components and systems when spawning a world to specify which access control implementation the world will have